comment-updater-headless/src/main/kotlin/PluginRunner.kt

```diff
@@ -0,0 +1,281 @@
+        statsHandler.refresh()
+    }
+
+
+    private fun collectProjectExamples(projectPath: String) {
+        val project = ProjectUtil.openOrImport(projectPath, null, true)
+        if( project == null) {
+            log(LogLevel.WARN, "Can't open project $projectPath")
+            return
+        }
+
+        val vcsManager = ServiceManager.getService(
+            project,
+            ProjectLevelVcsManager::class.java
+        ) as ProjectLevelVcsManagerImpl
+
+        val gitRepoManager = ServiceManager.getService(
+            project,
+            GitRepositoryManager::class.java
+        )
+
+        // Checkout https://intellij-support.jetbrains.com/hc/en-us/community/posts/206105769-Get-project-git-repositories-in-a-project-component
+        // To understand why we should call runAfterInitialization and create this runnable
+
+        val latch = CountDownLatch(1)
+        val runnable = {
+            try {
+                val gitRoots = vcsManager.getRootsUnderVcs(GitVcs.getInstance(project))
+                for (root in gitRoots) {
+                    val repo = gitRepoManager.getRepositoryForRoot(root) ?: continue
+                    runBlocking {
+                        val commits = repo.walkAll()
+                        statsHandler.numberOfCommits = commits.size
+
+                        commits.map { commit ->
+                            async(Dispatchers.Default) {
+                                processCommit(commit, project)
+                            }
+                        }.awaitAll()
+                    }
+
+                }
+            } catch (e: Exception) {
+                log(LogLevel.ERROR, "Failed with an exception: $e")
+                e.printStackTrace()
+            }
+            finally {
+                latch.countDown()
+            }
+        }
+
+        vcsManager.runAfterInitialization(runnable)
+        latch.await()
+    }
+
+
+    private fun processCommit(
+        commit: GitCommit,
+        project: Project
+    ) {
+        try {
+            commit.filterChanges(".java").forEach { change ->
+                val fileName = change.afterRevision?.file?.name ?: ""
+                log(
+                    LogLevel.INFO,
+                    "Commit: ${commit.id.toShortString()} num ~ ${statsHandler.processedCommits.get()}" +
+                            "/${statsHandler.numberOfCommits} File changed: $fileName",
+                    logThread = true
+                )
+
+                collectChange(change, commit, project)
+            }
+            statsHandler.processedCommits.incrementAndGet()
+        } catch (e: Exception) {
+            // In case of exception inside commit processing, just continue working and log exception
+            // We don't want to fall because of strange mistake on single commit
+            log(LogLevel.ERROR, "Error during commit ${commit.id.toShortString()} processing", logThread = true)
+            e.printStackTrace()
```

Comment:
please, remove `e.printStackTrace()`
if you want to catch the details, pass e.getMessage to the logger